packages/protocol/test/tokenomics/proofReward.test.ts

```diff
@@ -0,0 +1,293 @@
+import { expect } from "chai";
+import { BigNumber, ethers } from "ethers";
+import EventEmitter from "events";
+import { TaikoL1 } from "../../typechain";
+import { TestTkoToken } from "../../typechain/TestTkoToken";
+import { randEle } from "../utils/array";
+import { BlockInfo } from "../utils/block_metadata";
+import { BLOCK_PROPOSED_EVENT, BLOCK_PROVEN_EVENT } from "../utils/event";
+import { newProposerListener } from "../utils/propose";
+import Proposer from "../utils/proposer";
+import { newProverListener } from "../utils/prove";
+import Prover from "../utils/prover";
+import { createAndSeedWallets } from "../utils/seed";
+import sleep from "../utils/sleep";
+import {
+    sleepUntilBlockIsVerifiable,
+    verifyBlockAndAssert,
+    verifyBlocks,
+} from "../utils/verify";
+import { initTokenomicsFixture } from "./utils";
+
+describe("tokenomics: proofReward", function () {
+    let taikoL1: TaikoL1;
+    let l2Provider: ethers.providers.JsonRpcProvider;
+    let l1Signer: any;
+    let proposerSigner: any;
+    let proverSigner: any;
+    let genesisHeight: number;
+    let tkoTokenL1: TestTkoToken;
+    let interval: any;
+
+    beforeEach(async () => {
+        ({
+            taikoL1,
+            l2Provider,
+            l1Signer,
+            proposerSigner,
+            proverSigner,
+            genesisHeight,
+            tkoTokenL1,
+            interval,
+        } = await initTokenomicsFixture());
+    });
+
+    afterEach(() => clearInterval(interval));
+
+    it(`proofReward is 1 wei if the prover does not hold any tkoTokens on L1`, async function () {
+        const { maxNumBlocks, commitConfirmations } = await taikoL1.getConfig();
+
+        const proposer = new Proposer(
+            taikoL1.connect(proposerSigner),
+            l2Provider,
+            commitConfirmations.toNumber(),
+            maxNumBlocks.toNumber(),
+            0,
+            proposerSigner
+        );
+
+        const prover = new Prover(taikoL1, l2Provider, proverSigner);
+
+        let failedAssertion: Error | null = null;
+
+        const eventEmitter = new EventEmitter();
+
+        l2Provider.on(
```

Comment:
shall we add `l2Provider.off("block");` at the end of this test case? not sure if this is necessary, or if we can simply add it to `afterEach`.

Reply:
added to afterEach